.github/workflows/release_github.yml

```diff
@@ -4,33 +4,79 @@ on:
   workflow_dispatch:
     inputs:
       version_level:
-        description: "Semantic version level increase."
+        description: Semantic version level increase
         required: true
         type: choice
+        default: patch
         options:
           - patch
           - minor
           - major
 
 permissions:
   contents: write
-  pull-requests: write
   packages: read
+  statuses: read
+  checks: read
+  pull-requests: write
+  actions: read
+  repository-projects: read
+
+env:
+  GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  GH_PAGER: cat
 
 jobs:
+  checks:
+    name: Check requirements
+    runs-on: ubuntu-latest
+    steps:
+      - name: Fail if main branch was selected
+        if: ${{ github.ref_name == 'main' }}
+        run: |
+          echo "Cannot release from main branch, please select valid release branch."
+          exit 1
+
+      - name: Checkout repository
+        if: ${{ github.ref_name != 'develop' }}
+        uses: actions/checkout@v4
+
+      - name: Check if PR exists
+        if: ${{ github.ref_name != 'develop' }}
+        run: gh pr view ${{ github.ref_name }}
+
+      - name: Check if PR base is main
+        if: ${{ github.ref_name != 'develop' }}
+        run: gh pr view ${{ github.ref_name }} --json baseRefName -q 'if .baseRefName == "main" then "PR base is main" else error("PR base is not main") end'
+
+      - name: Check if PR is mergeable
```

Comment:
Shouldn't this check happen here, also if the release branch is develop? See also [this comment](https://github.com/EIT-ALIVE/eitprocessing/pull/338#discussion_r1890369062)

Reply:
If the release branch is develop, there is no PR. This only works on PRs. 

If the release branch is develop, there are no checks. These checks are done before PRs are merged into develop. As long as that is always the case, there are no checks required for develop.

Alternatively, we could also create a PR for merging develop into main. That would simplify this workflow a bit, but also add a step when creating a release.

Reply:
> If the release branch is develop, there are no checks. These checks are done before PRs are merged into develop. As long as that is always the case, there are no checks required for develop.

I think there is no formal restriction for develop to have conflicts to main. This could either happen due to non-careful PR review, or e.g. after a hotfix was made directly to main.


> Alternatively, we could also create a PR for merging develop into main. That would simplify this workflow a bit, but also add a step when creating a release. I don't dislike the idea, though. The workflow for making a release would change to:
> 
>   * create PR (either from branch or develop)
>   * trigger action

I also like this workflow. Also makes things more consistent

Reply:
Agreed. I implemented this in https://github.com/EIT-ALIVE/eitprocessing/pull/338/commits/379316bcbf3fd0cc6a67868e66670d301e135788.

Reply:
in fact, the documentation already says to this: [step 0](https://github.com/EIT-ALIVE/eitprocessing/blob/main/README.dev.md#automated-release-workflow) :D